src/camera.js

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+import initModule from "../build/libapi.mjs";
+
+export function rethrowIfCritical(err) {
+  // If it's precisely Error, it's a custom error; anything else - SyntaxError,
+  // WebAssembly.RuntimeError, TypeError, etc. - is treated as critical here.
+  if (err.constructor !== Error) {
+    throw err;
+  }
+}
+
+const INTERFACE_CLASS = 6; // PTP
+const INTERFACE_SUBCLASS = 1; // MTP
+
+let ModulePromise = null;
+
+class Camera {
+  constructor() {
+    this.queue = Promise.resolve();
+    this.Module = null;
+    this.context = null;
+  }
+
+  async showCameraPicker() {
```

Comment:
Sorry, one more nit: this should be either standalone function or perhaps better yet a static method, as it's not tied to the instance. For example:

```suggestion
  static async showPicker() {
```

so user can call it as `await Camera.showPicker()`.